### DISCUSSION TITLE
Public part of DB licensing / Лицензирование публичной части БД

### DISCUSSION THREAD

**Discussion by mkgrgis** [PastVu/pastvu#799] [Ideas / Идеи и предложения]
I think, PastVu team must determine a license for public part of DB as for data not about users, not representing user's comments and not content of images. Now as public data i meant geodata of images, metadata of images, regoin's data including borders.
[ODbL](https://web.archive.org/web/20171010115007/https://opendatacommons.org/licenses/odbl/1.0/) can be adequate to most of outer API applications. 

Предлагается выбрать лицензию на публичную часть БД PastVu, к которой не относятся данные о пользователях(кроме ника), их комментарии и содержание изображений. К публичной части на данным момент предлагаю отнести координаты и полное описание изображения (включая адрес публично доступного файла) а также все данные регионов. В качестве первого варианта лицензирования предлагаю определить лицензию [ODbL](https://web.archive.org/web/20171126065952/http://gis-lab.info/blog/2010-06/odbl-v1-russian/).
Недавно написал прогармму-архиватор через PasVu API и хочу иметь ясность по использованию самих JSON данных, охватывающих каждое изображение. На случай проблем с ресурсом это наиболее ценная часть проекта, так как картинки можно найти в различных службах архивирования, а связаные данные едва ли.

**Comment by kabalin** [comment]
Summary of ODbL terms (https://opendatacommons.org/licenses/odbl/summary/):

> You are free:
> 
> * To share: To copy, distribute and use the database.
> * To create: To produce works from the database.
> * To adapt: To modify, transform and build upon the database.
>
> As long as you:
> * Attribute: You must attribute any public use of the database, or works produced from the database, in the manner specified in the ODbL. For any use or redistribution of the database, or works produced from it, you must make clear to others the license of the database and keep intact any notices on the original database.
> * Share-Alike: If you publicly use any adapted version of this database, or works produced from an adapted database, you must also offer that adapted database under the ODbL.
> * Keep open: If you redistribute the database, or an adapted version of it, then you may use technological measures that restrict the work (such as DRM) as long as you also redistribute a version without such measures.
> 

It makes things interesting in fact.

1. OSM is providing its data under ODbL, this mean we should attribute use of OSM for regions data already (https://wiki.osmfoundation.org/wiki/Licence/Attribution_Guidelines). The simplest probably would be to add information to "About" modal: "Administrative and geographic boundaries data: © OpenStreetMap contributors".

2. Test database we provide for developers is containing mainly regions, thus should probably be ODbL licensed too (per [FAQ](https://wiki.openstreetmap.org/wiki/License/Use_Cases#Case_3:_I_want_to_publish_something_based_on_OSM_and_my_own_data)).

3. I am not sure about contributed data (photo data and locations), if we treat them similar way that OSM does, the user who is adding data to the map should agree that data will be used and licensed under the terms of ODbL. This excludes image itself which should always be treated as copyrighted I suppose.

**Comment by mkgrgis** [comment]
I support these conclusions @kabalin ! That is exactly what was meant.
I consider the main technical problem is to include license string into API data, into JSON response itself. OSM API responses always contain a license string.

> This excludes image itself which should always be treated as copyrighted I suppose.

I think user's comments and user's data excluded nickname must be always be treated as copyrighted, not ODbL.

**Comment by kabalin** [comment]
So far, points 1 and 2 from [comment above](https://github.com/PastVu/pastvu/issues/402#issuecomment-1022713210) has been addressed (added relevant attribution and ODbL mention in readme file, "about" page and API landing page).

With regard to photo coordinates data returned in API, I don't think we should do anything at this point, as this would mean every user who uploaded the photo in the past should agree to ODbL terms (in my understanding), which does not seem feasible. For now I would suggest to treat all photo data as copyrighted and close the issue. We can get back to this discussion when have a strong usecase to action upon.

**Comment by mkgrgis** [comment]
> We can get back to this discussion when have a strong usecase to action upon.

What about retromap.ru and [this program](https://github.com/mkgrgis/PastVu_geo_archiv) ? Why only version about ODbL   for photo metadata?
